src/mdacli/libcli.py

```diff
@@ -246,6 +247,54 @@ def add_output_group(analysis_class_parser):
     )
 
 
+def _create_mda_file_completer(format_dict):
```

Comment:
I think you don't need the whole dictionary but just the list of keys. Maybe passing a list of file extension makes this function a bit clearer

```suggestion
def _create_extension_completer(extension_list):
```

We could also make this public as this might be useful in general.

Reply:
If you loop through a dictionary what you get are the keys.